### PR TITLE
[Tai Le] Sorting

### PR DESCRIPTION
**Pull request by Ta1grr**
**Objective**
 - [x] iterative-sorting.py
 - [x] recursive-sorting.py

**Reviewer**
@luisansuarez

### CODE REVIEW

File: project/iterative_sorting.py [hunk 1]
```diff
@@ -6,13 +6,14 @@ def selection_sort( arr ):
...
-                    smallest_index = j
+        for j in range(cur_index, len(arr)):
+            if arr[j] < arr[smallest_index]:
+                smallest_index = j
 
 
-        # TO-DO: swap
 
+        # TO-DO: swap
+        arr[cur_index], arr[smallest_index] = arr[smallest_index], arr[cur_index]
```
Comment (LuisanSuarez) [inline project/iterative_sorting.py:17]: nice way to swap!

**Comment by LuisanSuarez** [comment]
make sure you turn insertion sort in today! 
also, please leave comments above each line explaining what they do. It's a bit tedious, I know, but it'll force you to understand exactly what it is your doing in plain english.

File: project/iterative_sorting.py [hunk 2]
```diff
@@ -23,11 +23,27 @@ def selection_sort( arr ):
 # TO-DO: implement the Insertion Sort function below
 def insertion_sort( arr ):
     for i in range(1, len(arr)):
-        cur_value = 
-        while 
+        cur_value = None
+        for  j in range(i, len(arr[::-i]), -1):
+            j > cur_value
```
Comment (LuisanSuarez) [inline project/iterative_sorting.py]: this doesn't look quite finished to me. you're doing a comparison, but no swapping. I also get errors straight away when running this.

File: project/recursive_sorting.py [hunk 3]
```diff
@@ -51,6 +51,43 @@ def quick_sort( arr, low, high ):
         quick_sort(high)
     return arr
 
+#Quick Sort example we used during class:
+def quick_sort(a):
```
Comment (LuisanSuarez) [inline project/recursive_sorting.py]: I ran this code and it also doesn't seem to work. Did you test this?

File: project/iterative_sorting.py (hunk 2 above)
Comment (Ta1grr) [reply project/iterative_sorting.py]: Just commit the latest version, forgot to update it.

**Comment by LuisanSuarez** [comment]
Tai, it looks like you're very well on the right track but you didn't quite finish the algorithms. Are you running tests on these before you commit? If so, then reach out to me please as for some reason they're not running when I run them on my machine. But if you're not, make sure you run the program and get the results you want _before_ committing.
